src/components/Select/Select.js

```diff
@@ -157,20 +161,31 @@ class Select extends React.PureComponent {
       >
         {downshiftProps => {
           const { isOpen, inputValue, itemToString, selectedItem } = downshiftProps;
+          /* if select is closed */
           if (!isOpen) {
+            /* update justOpened to false, to know when exactly it will change to true first time */
             justOpened = isOpen;
+            /* downshift doesn't handle the changes inside of the `itemToString` function, so if your data
```

Comment:
I know that Downshift not knowing about changes to `itemToString` is part of the behavior you're trying to address, but I still feel like this is a very complicated solution, when we could just as easily put the Select in a `loading` state until all the data needed to render the items is loaded. It's actually a better user experience to not show the data until it is ready, to avoid the data suddenly changing in a confusing way. I'd prefer to take the simpler solution than introduce complex logic in this component for a particular use case and risk it breaking for others.